gnovm/pkg/gnolang/nodes.go

```diff
@@ -627,36 +726,60 @@ type ChanTypeExpr struct {
 	Value Expr    // value type
 }
 
+func (x *ChanTypeExpr) addressability() Addressability {
+	return AddressabilityNotApplicable
+}
+
 type FuncTypeExpr struct {
 	Attributes
 	Params  FieldTypeExprs // (incoming) parameters, if any.
 	Results FieldTypeExprs // (outgoing) results, if any.
 }
 
+func (x *FuncTypeExpr) addressability() Addressability {
+	return AddressabilityNotApplicable
```

Comment:
For these cases where we just return NotApplicable, I wonder if we should just panic...

Reply:
In the case of type expressions, yes, we should panic because it is unexpected to have this method called https://github.com/gnolang/gno/pull/2731/commits/8bec242b7beacc6d3bb4f658255390671fadf505

But we shouldn't get rid of NotApplicable altogether. There are some valid cases, for example, when there is code like this:
```go
func main() {
    _ = &DoSomething()
}

func DoSomething() (int, int) { ... }
```
In this case addressability is not applicable because it should fail with an error message referencing the fact that it is trying to take a reference, which requires one operand, on the result of a function that returns multiple values, so we don't want to trigger an addressability error here. In Go, this would panic with something like `multiple-value DoSomething() (value of type (int, int)) in single-value context`.